### PR TITLE
v0.4.x hardening: edge-case contract, entry validation, native exponential

### PR DESCRIPTION
**Pull request by DougManuel**
Phase 1 of the post-v0.4.0 development plan (`development/post-v040-development-plan.md`). Four tasks, each TDD with per-task review, plus a final whole-branch review.

## Changes

- **Edge-case contract tests** (#35) — new `tests/testthat/test-edge-case-contract.R` pins verified behaviour: zero-row/zero-column generation, n validation messages, loud zero-metadata failure, single-row metadata. Evidence of the 6-vs-11-column fallback discrepancy recorded on #14. No production-code changes.
- **Entry-validation alignment** (#36) — `create_mock_data()` now accepts `n = 0` (full-schema zero-row frame), rejects fractional/negative/NA `n` with the spec layer's message (previously `n = 1.5` was silently accepted), and fails upfront with a named error when `databaseStart` is missing.
- **Native exponential distribution** (#37) — `rate` parameter on both continuous constructors, validator rule, inverse-CDF range-truncated generator (documented difference from the unbounded legacy `rexp()`), recodeflow adapter rate-forwarding, and orchestrator allowlist update so exponential metadata routes natively instead of falling back to the legacy engine. Covered end-to-end including a recodeflow-metadata integration test.

## Review notes

- Final review verdict: ready to merge. One API decision made during review: `rate` sits after `rtype` in the constructor signatures to preserve positional compatibility.
- Adapter change verified backward-compatible for downstream metadata without a `rate` column (cchsflow/chmsflow).
- Follow-up filed: #44 (`is.finite(n)` in both guards).
- Suite at branch head: FAIL 0 | WARN 35 (pre-existing) | SKIP 10 (pre-existing) | PASS 673.

Closes #35. Closes #36. Closes #37.

### CODE REVIEW

File: R/create_mock_data.R [hunk 1]
```diff
@@ -237,8 +245,9 @@ create_mock_data <- function(databaseStart,
 
   # ========== VALIDATE INPUT ==========
 
-  if (n < 1) {
-    stop("n must be at least 1")
+  if (!is.numeric(n) || length(n) != 1 || is.na(n) ||
+      n < 0 || n != trunc(n)) {
+    stop("n must be a non-negative whole number.", call. = FALSE)
   }
```
[the remark this answers is not in the document]
Comment (DougManuel) [reply R/create_mock_data.R]: Fixed in 361e269 — both guards (create_mock_data() entry and validate_mock_spec()) now reject non-finite n with the shared message, and Inf was added to the rejection tests at both layers. This was tracked as #44; the commit closes it on merge.

File: R/mock_spec_native.R [hunk 2]
```diff
@@ -105,6 +105,23 @@
   values
 }
 
+#' @noRd
+.native_truncated_exponential <- function(n, rate, range, variable_name) {
+  lower <- range[[1]]
+  upper <- range[[2]]
+  p_lower <- stats::pexp(lower, rate = rate)
+  p_upper <- stats::pexp(upper, rate = rate)
+  if (!is.finite(p_lower) || !is.finite(p_upper) || p_upper <= p_lower) {
```
[the remark this answers is not in the document]
Comment (DougManuel) [reply R/mock_spec_native.R:121]: Fixed in 361e269 — .native_truncated_exponential() now returns numeric(0) for n = 0 before the no-mass guard, matching the zero-row contract pinned in test-edge-case-contract.R. Added a test: n = 0 with a degenerate range (c(5,5)) returns an empty typed column instead of erroring.
